### ISSUE TITLE
Expired SP Password - Cluster not working at all anymore

### ISSUE THREAD

**Issue by DonMartin76** [Azure/aks-engine#1229]
**Is this a request for help?**: Yes-ish

---

**Is this an ISSUE or FEATURE REQUEST?** (choose one): Issue

---

**What version of aks-engine?**: 0.33.4

---

**Kubernetes version**: 1.13.5

**What happened**: The password/client secret for the service principal we used to create the cluster expired, which entrailed a couple of unwanted effects. At least we *think* that this was the cause of the issues:

* Updates to the LoadBalancer could no longer be done (we could see that the controller manager was failing; this was how we saw that the password had expired)
* No pods were rescheduled/restarted after they were killed - this was a really unexpected effect; deleting a pod should obviously re-launch a new one due to the deployment manifests, but this **did not happen**

We tried to "resurrect" the cluster by following the instructions from https://github.com/Azure/aks-engine/issues/724, but we never managed to get the cluster back up in a working state. The solution in the end was to set up a new cluster (using the new credentials). This was not as bad as it could have been as the affected cluster was only a staging cluster, but had this happened with the production cluster, it would have been a lot worse.

**What you expected to happen**: This is a tough question. I would hope that I could update the client secret/password somehow, preferably as easily as possible. And/or that I would be warned beforehand that a credential is about to expire.

**How to reproduce it** (as minimally and precisely as possible): This is also a tough one, but it happened to us where a change in the cluster had an impact on the LoadBalancer setting, but the service principal client secret had expired.

In short: For us, this is not an immediate call for action from your side in any way, this is more to document what might happen if the client secret expires in a running cluster.

**Comment by CecileRobertMichon** [comment]
related to https://github.com/Azure/aks-engine/issues/724 and #1189

**Comment by andyzhangx** [comment]
I think you could renew the original password, that's the easiest way to make it work. you could follow this: https://github.com/andyzhangx/demo/blob/master/issues/sp-issues.md#option1-extend-the-original-sp-password-for-one-more-year

**Comment by DonMartin76** [comment]
@andyzhangx I followed exactly those instructions, but I still couldn't make the cluster work again. It seemed to be "dead"; no matter which actions I did, restarting the containers directly on the master etc., I couldn't make it work again. Eventually I gave up and recreated the entire thing instead.

Now, it's not 100% sure that the SP was the reason for this, but the indications are pretty strong, in my opinion.

**Comment by andyzhangx** [comment]
@DonMartin76 then I think you should get kubelet log first to check what happend: `sudo journalctl -u kubelet -l > kubelet.log`, it should tell you what happened.

**Comment by DonMartin76** [comment]
@andyzhangx Unfortunately, I have deleted the cluster. I could have thought of this. But I didn't.
